lib/pbench/agent/tool_meister.py

```diff
@@ -50,23 +50,59 @@
 from distutils.spawn import find_executable
 from pathlib import Path
 
-import daemon
 import pidfile
 import redis
 
-from pbench.common.utils import md5sum
-import pbench.agent.toolmetadata as toolmetadata
+from daemon import DaemonContext
+from redis.connection import SERVER_CLOSED_CONNECTION_ERROR
 
+import pbench.agent.toolmetadata as toolmetadata
 
-# Path to external tar executable.
-tar_path = None
+from pbench.common.utils import md5sum
 
-# Path to external pbench-sysinfo-dump executable.
-sysinfo_dump = None
 
 # FIXME: The client response channel should be in a shared constants module.
 client_channel = "tool-meister-client"
 
+# Logging format string for unit tests
+fmtstr_ut = "%(levelname)s %(name)s %(funcName)s -- %(message)s"
+fmtstr = "%(asctime)s %(levelname)s %(process)s %(thread)s %(name)s %(funcName)s %(lineno)d -- %(message)s"
+
+
+class RedisHandler(logging.Handler):
+    """Publish messages to a given channel on a Redis server.
+    """
+
+    def __init__(
+        self,
+        channel,
+        hostname=None,
+        redis_client=None,
+        level=logging.NOTSET,
+        **redis_kwargs,
+    ):
+        """Create a new logger for the given channel and redis_client.
+        """
+        super().__init__(level)
+        self.channel = channel
+        self.hostname = hostname
+        self.redis_client = redis_client or redis.Redis(**redis_kwargs)
+        self.counter = 0
+        self.errors = 0
+
+    def emit(self, record):
+        """Publish record to redis logging channel
+        """
+        try:
+            formatted_record = self.format(record)
+            self.redis_client.publish(
+                self.channel, f"{self.hostname} {self.counter:04d} {formatted_record}"
```

Comment:
Zero padded counter makes combined log messages a bit more sortable (at least until you have 10,000 or more).